File: 1449-print-words-vertically/1449-print-words-vertically.py
class Solution:
    def printVertically(self, s: str) -> List[str]:
        res=" "
        x=s.split()
        # print(s)
        # print(len(s))
        c=0
        j=0
        ans=[]
        ma,mi=0,0
        for i in x:
            ma=max(len(i),ma)
            mi=min(len(i),mi)
        # print(ma,mi)
        while(c<ma):
            res=[]
            for i in range(len(x)):
                if c<len(x[i]) and x[i][c]!=" ":
                    res.append(x[i][c])       
                else:
                    res.append(" ")
            c+=1
            if c>mi:
                while(res):
                    if res[-1]==" ":
                        res.pop()
                    else:
                        break
            ans.append("".join(res))
        # print(ans)
        return ans
        

File: 1449-print-words-vertically/1449-print-words-vertically.java
class Solution {
    public List<String> printVertically(String s) {
       String[] array = s.split(" ");
        int i=0;
        String res=" ";
        int mal=0;
        int mil=999;
        for(String str : array)
        {
            mal =Math.max(str.length(),mal);
            mil =Math.min(str.length(),mil);
        }
        ArrayList<String> result= new ArrayList<>();
        for(i=0;i<mal;i++)
        {
            res="";
            for(int j=0;j<array.length;j++)
            {
                if(i<array[j].length())
                {
               
                String word=array[j];
                res+=word.charAt(i);
                }
                else{
                  res+=" ";
                }
            }
            int m=0;
            m=res.length();
            int k=m-1;
            for(k=m-1;k>=0;k--){
                if (res.charAt(k)!=' ')
                break;
            }

            res=res.substring(0,k+1);
            result.add(res);
        }
        return result;
    }
}